pkg/generate/app/cmd/newapp.go

```diff
@@ -262,7 +263,12 @@ func (c *AppConfig) validate() (app.ComponentReferences, app.SourceRepositories,
 		errs = append(errs, fmt.Errorf("when --strategy is specified you must provide at least one source code location"))
 	}
 
-	env, duplicateEnv, envErrs := cmdutil.ParseEnvironmentArguments(c.Environment)
+	env, duplicateEnv, envErrs := cmdutil.Environment{}, []string{}, []error{}
+	if c.BuildEnvironment != nil {
+		env, duplicateEnv, envErrs = cmdutil.ParseEnvironmentArguments(c.BuildEnvironment)
+	} else {
+		env, duplicateEnv, envErrs = cmdutil.ParseEnvironmentArguments(c.Environment)
+	}
```

Comment:
Wouldn't it make sense to just leave this alone since you are setting BuildEnv = Environment here? https://github.com/openshift/origin/pull/4463/files#diff-67f42376bdba60c035c6268d160059baR625